plasmapy/atomic/exceptions.py

```diff
@@ -0,0 +1,57 @@
+class AtomicError(Exception):
+    """An exception for errors in the `~plasmapy.atomic` subpackage."""
+    pass
+
+
+class MissingAtomicDataError(AtomicError):
+    """An exception for missing atomic or particle data."""
+    pass
+
+
+class ChargeError(AtomicError):
+    """An exception for incorrect or missing charge information."""
+    pass
+
+
+class UnexpectedParticleError(AtomicError):
+    """An exception for when a particle is not of the expected category."""
+    pass
+
+
+class InvalidIonError(UnexpectedParticleError):
+    """
+    An exception for when an argument is a valid particle but not a
+    valid ion.
+    """
+    pass
+
+
+class InvalidIsotopeError(UnexpectedParticleError):
+    """
+    An exception for when an argument is a valid particle but not a
+    valid isotope.
+    """
+    pass
+
+
+class InvalidElementError(UnexpectedParticleError):
+    """
+    An exception for when an argument is a valid particle is not a
+    valid element.
+    """
+    pass
+
+
+class InvalidParticleError(AtomicError):
+    """An exception for when a particle is invalid."""
+    pass
+
+
+class AtomicWarning(Warning):
+    """The base warning for the `~plasmapy.atomic` subpackage."""
+    pass
+
+
+class MissingAtomicDataWarning(AtomicWarning):
+    """Warning for use when atomic or particle data is missing."""
+    pass
```

Comment:
Yeah, this is a much better place for `atomic` exceptions than in `utils`, so thank you for making these changes.

One minor change here would be to add a newline to the end of the file.

Reply:
I've been trying to mess around with `autopep8` and our `pycodestyle` settings in `setup.cfg` to make adding those automatic, but haven't been successful yet.